BiocManager::install("tximeta")

library(tximeta)
library(SummarizedExperiment)
library(readxl)
library(GenomicFeatures)


## add annotation for the geneIDs

# point to a Salmon quantification file with an additional artificial transcript
dir <- "output_JJ1715_393_S43_R1_001.fastq_paired.fq_quant/"
file <- file.path(dir, "quant.sf")
coldata <- data.frame(files=file, names="JJ1715_393_S43", sample="1",
                      stringsAsFactors=FALSE)

# now point to the Salmon index itself to create a linkedTxome
# as the index will not match a known txome
indexDir <- file.path("/home/julianne/Documents/transcriptomicsonhoffman/Mus_musculus_c57bl6nj_index/")

# point to the source FASTA and GTF:
fastaFTP <- c("ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz",
              "ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz",
              "extra_transcript.fa.gz")
gtfPath <- file.path(dir, "Drosophila_melanogaster.BDGP6.22.98.plus.gtf.gz")

# now create a linkedTxome, linking the Salmon index to its FASTA and GTF sources
makeLinkedTxome(indexDir=indexDir, source="Ensembl", organism="Mus musculus",
                release="98", genome="BDGP6.22", fasta=fastaFTP, gtf=gtfPath, write=FALSE)


## Create a DESeqDataSet object
dds <- DESeqDataSetFromTximport(txi, sampleTable, ~ condition)
